pkg/gadgets/trace/dns/tracer/latency.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	"time"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+const dnsLatencyCacheSize int = 1024
+
+// dnsQueryKey is a unique identifier for a DNS query.
+// The address is the query's source IP address (equals destination address of the response).
+// The ID comes from the DNS header.
+type dnsQueryKey struct {
+	addr [16]uint8 // Either IPv4 or IPv6.
+	id   uint16
+}
+
+// dnsLatencyCalculator calculates the latency between a query and its response.
+// It uses an LRU cache to bound memory usage.
+// All operations are thread-safe.
+type dnsLatencyCalculator struct {
+	queryCache *lru.Cache[dnsQueryKey, uint64] // This is thread-safe.
+}
+
+func newDNSLatencyCalculator() (*dnsLatencyCalculator, error) {
+	queryCache, err := lru.New[dnsQueryKey, uint64](dnsLatencyCacheSize)
```

Comment:
Just one curiosity, I check the documentation, and it seems there is no need to clean/remove the lru once we are done. It is all left to the GC?

Reply:
yep, that's my understanding too -- should get cleaned up by the garbage collector